### PR TITLE
PRIME-1952 Performance improvement enrollee overview

### PR DESCRIPTION
**Pull request by neophyte57**
Should not see these errors for non-paper submission enrollees in Admin's Enrollee Overview screen:

image

### CODE REVIEW

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts [hunk 1]
```diff
@@ -168,4 +173,8 @@ export class EnrolleeOverviewComponent extends AdjudicationContainerComponent im
       ...remainder
     };
   }
+
+  private isPaperEnrollee(enrollee: Enrollee): boolean {
+    return (enrollee?.gpid && enrollee.gpid.startsWith('NOBCSC'));
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts:178]: could utilize the `const PAPER_ENROLLEE_GPID_PREFIX` in `lib/constants.ts`

File: prime-dotnet-webapi/Controllers/EnrolleePaperSubmissionsController.cs [hunk 2]
```diff
@@ -263,11 +263,6 @@ public async Task<ActionResult> SetEnrolleePaperSubmissionProfileCompleted(int e
         [ProducesResponseType(typeof(ApiResultResponse<EnrolleeAdjudicationDocument>), StatusCodes.Status200OK)]
         public async Task<ActionResult> GetAdjudicationDocuments(int enrolleeId)
         {
-            if (!await _enrolleePaperSubmissionService.PaperSubmissionIsEditableAsync(enrolleeId))
-            {
-                return NotFound($"No Editable Paper Submission found with Enrollee Id {enrolleeId}");
-            }
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Controllers/EnrolleePaperSubmissionsController.cs:269]: If not checking enrollee existence any more, should also remove 
 attribute `[ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]`

Comment (james-hollinger) [reply prime-dotnet-webapi/Controllers/EnrolleePaperSubmissionsController.cs:269]: Should probably check for existence of the `Enrollee` only, not if it's editable

**Review by johnnyl6 — APPROVED** [review]
LGTM

File: prime-dotnet-webapi/Controllers/EnrolleePaperSubmissionsController.cs [hunk 3]
```diff
@@ -263,9 +263,9 @@ public async Task<ActionResult> SetEnrolleePaperSubmissionProfileCompleted(int e
         [ProducesResponseType(typeof(ApiResultResponse<EnrolleeAdjudicationDocument>), StatusCodes.Status200OK)]
         public async Task<ActionResult> GetAdjudicationDocuments(int enrolleeId)
         {
-            if (!await _enrolleePaperSubmissionService.PaperSubmissionIsEditableAsync(enrolleeId))
+            if (!await _enrolleeService.EnrolleeExistsAsync(enrolleeId))
```
Comment (hogmann) [inline prime-dotnet-webapi/Controllers/EnrolleePaperSubmissionsController.cs:266]: Was this intentional? Should this be checking if it is a paper enrollee not just an enrollee in general

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts [hunk 4]
```diff
@@ -114,8 +112,15 @@ export class EnrolleeOverviewComponent extends AdjudicationContainerComponent im
...
-      ).subscribe();
+            catchError(_ => of([])))),
+        exhaustMap(() =>
+          this.isPaperEnrollee(this.enrollee) ?
+            this.paperEnrolmentResource.getAdjudicationDocuments(+this.route.snapshot.params.id) :
+            of(null))
+      ).subscribe(
+        // Note that these documents are currently not displayed for a Completed Paper Submission
+        // due to how `showAdjudication` is set and causing DocumentAttachmentsComponent to be hidden
+        documents => this.documents = documents);
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts:123]: Missing the typing on documents. As well as what looks like some formatting

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts [hunk 5]
```diff
@@ -168,4 +173,8 @@ export class EnrolleeOverviewComponent extends AdjudicationContainerComponent im
       ...remainder
     };
   }
+
+  private isPaperEnrollee(enrollee: Enrollee): boolean {
+    return (enrollee?.gpid && enrollee.gpid.startsWith(PAPER_ENROLLEE_GPID_PREFIX));
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts:178]: Could do this without the &&

``` return enrollee?.gpid?.startsWith(PAPER_ENROLLEE_GPID_PREFIX); ```

**Review by AMiKaH — APPROVED** [review]

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts (hunk 4 above)
Comment (neophyte57) [reply prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts]: What's the formatting issue(s)?

Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts]: documents is not typed
``` subscribe((documens: type) => this.documents = documents) ```
Also not sure what the comment represents in this case, maybe jsut add a TODO: display documents for completed paper submission

Comment (neophyte57) [reply prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts]: I removed comment and created a Jira ticket instead:  PRIME-2038

**Review by hogmann — APPROVED** [review]
LGTM

**Pull request merged**